### ISSUE TITLE
redefine PadListNames in the c template might be useless in 5.22

### ISSUE THREAD

**Issue by atoomic** [CpanelInc/perl-compiler#16]
need to check if it s required in the current perl 5.22 we are using

#undef PadListNames
#define PadListNames

**Comment by atoomic** [comment]
lib/B/C/Templates/base.c.tt2:148:#define PadlistNAMES(pl)       *((PADNAMELIST **)PadlistARRAY(pl))

**Comment by atoomic** [comment]
the change was introduced by 9b7476d7 and is available since v5.21.7

commit 9b7476d7a269a4d9bb24393ae5c8d75efe2fcab4
Author: Father Chrysostomos <sprout@cpan.org>
Date:   Sun Nov 23 14:25:22 2014 -0800

    Make PADNAMELIST a separate type

    This is in preparation for making PADNAME a separate type.

diff --git a/pad.h b/pad.h
-#define PadlistNAMES(pl)       (*PadlistARRAY(pl))
+#define PadlistNAMES(pl)       ((PADNAMELIST *)*PadlistARRAY(pl))

**Comment by atoomic** [comment]
this is still required... we cannot remove that change
`````
/* PadlistNAMES broken as lvalue with v5.21.6-197-g0f94cb1, fixed with 5.22.1 and 5.23.0 */
/* TODO: REMOVE ME after 5.22.1 is released. */
#undef PadlistNAMES
#define PadlistNAMES(pl)       *((PADNAMELIST **)PadlistARRAY(pl))
`````

this was fixed by commit

commit 73949fca082fe50bf47755c5ffa328259057ae36
Author: David Mitchell <davem@iabyn.com>
Date:   Mon Jun 8 09:15:17 2015 +0100

    make PadlistNAMES() lvalue again.

which is only in v5.23.0 and later
